lib/adapters/http.js

```diff
@@ -224,3 +244,6 @@ module.exports = function httpAdapter(config) {
     }
   });
 };
+
+
+module.exports._isANoProxyHost = isANoProxyHost;
```

Comment:
Please don't export functions just for testing. This can be tested throught the behaviour of the request.